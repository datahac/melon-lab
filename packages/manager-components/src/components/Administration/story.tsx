import { action } from '@storybook/addon-actions';
import { storiesOf } from '@storybook/react';
import React from 'react';
import Administration from './index';

const data = {
  toggleSubscription: action('clicked toggleSubscription'),
  convertUnclaimedRewards: action('clicked convertUnclaimedRewards'),
  shutdown: action('clicked shutdown'),
  quoteAsset: 'MLN',
};

storiesOf('Components|Administration', module).add('Default', () => {
  return <Administration {...data} />;
});
