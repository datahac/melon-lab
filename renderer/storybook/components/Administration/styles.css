.administration {
  max-width: 400px;
  margin: 0 auto var(--space-xl) auto;
}
