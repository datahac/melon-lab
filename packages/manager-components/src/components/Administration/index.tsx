import React, { StatelessComponent } from 'react';
import Button from '~/blocks/Button';

import styles from './styles.css';

export interface AdministrationProps {
  subscriptionAllowed?: boolean;
  toggleSubscription: () => void;
  convertUnclaimedRewards: () => void;
  shutdown: () => void;
  loading?: boolean;
  registerForCompetition?: () => void;
  fundAddress?: string;
  isCompeting?: boolean;
  quoteAsset: string;
}

export const Administration: StatelessComponent<AdministrationProps> = ({
  subscriptionAllowed,
  toggleSubscription,
  convertUnclaimedRewards,
  shutdown,
  loading,
  registerForCompetition,
  fundAddress,
  isCompeting,
  quoteAsset,
}) => {
  return (
    <div className="administration">
      <style jsx>{styles}</style>
      <h3>Fund Administration</h3>
      {/* {!isCompeting ? (
          <Button onClick={() => registerForCompetition(fundAddress)}>
            Register for competition
          </Button>
        ) : (
          ""
        )} */}
      {subscriptionAllowed ? (
        <Button style="clear" onClick={toggleSubscription}>
          Disable subscription
        </Button>
      ) : (
        <Button style="clear" onClick={toggleSubscription}>
          Enable subscription
        </Button>
      )}

      <Button style="clear" onClick={convertUnclaimedRewards}>
        Convert unclaimed rewards: 0 {quoteAsset}
      </Button>
      <Button style="clear" onClick={shutdown}>
        Irreversibly shut down fund
      </Button>
      {loading && <p>Please wait for upcoming Metamask popup</p>}
    </div>
  );
};

export default Administration;
