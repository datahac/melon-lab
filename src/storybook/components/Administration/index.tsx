import React, { StatelessComponent } from 'react';
import Button from '~/blocks/Button';

import styles from './styles.css';

export interface AdministrationProps {
  convertUnclaimedRewards: () => void;
  fundAddress?: string;
  isCompeting?: boolean;
  loading?: boolean;
  quoteAsset: string;
  registerForCompetition?: () => void;
  shutdown: () => void;
  subscriptionAllowed?: boolean;
  toggleSubscription: () => void;
}

export const Administration: StatelessComponent<AdministrationProps> = ({
  convertUnclaimedRewards,
  fundAddress,
  isCompeting,
  loading,
  quoteAsset,
  registerForCompetition,
  shutdown,
  subscriptionAllowed,
  toggleSubscription,
}) => (
  <div className="administration">
    <style jsx>{styles}</style>
    <h3>Fund Administration</h3>
    {/* {!isCompeting ? (
          <Button onClick={() => registerForCompetition(fundAddress)}>
            Register for competition
          </Button>
        ) : (
          ""
        )} */}
    {subscriptionAllowed ? (
      <Button style="clear" onClick={toggleSubscription}>
        Disable subscription
      </Button>
    ) : (
      <Button style="clear" onClick={toggleSubscription}>
        Enable subscription
      </Button>
    )}

    <Button style="clear" onClick={convertUnclaimedRewards}>
      Convert unclaimed rewards: 0 {quoteAsset}
    </Button>
    <Button style="clear" onClick={shutdown}>
      Irreversibly shut down fund
    </Button>
    {loading && <p>Please wait for upcoming Metamask popup</p>}
  </div>
);

export default Administration;
