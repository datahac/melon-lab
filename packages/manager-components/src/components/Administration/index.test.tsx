import React from 'react';
import Administration from './index';

const data = {
  toggleSubscription: () => null,
  convertUnclaimedRewards: () => null,
  shutdown: () => null,
  quoteAsset: 'MLN',
};

describe('Administration', () => {
  const defaultElement = <Administration {...data} />;

  it('should render correctly', () => {
    const wrapper = shallow(defaultElement);
    expect(wrapper).toMatchSnapshot();
  });
});
