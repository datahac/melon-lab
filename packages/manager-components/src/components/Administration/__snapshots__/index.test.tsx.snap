// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Administration should render correctly 1`] = `
<div
  className="jsx-__scopedHash administration"
>
  <JSXStyle
    css="__scoped"
    styleId="__scopedHash"
  />
  <h3
    className="jsx-__scopedHash"
  >
    Fund Administration
  </h3>
  <Button
    onClick={[Function]}
    style="clear"
  >
    Enable subscription
  </Button>
  <Button
    onClick={[Function]}
    style="clear"
  >
    Convert unclaimed rewards: 0 
    MLN
  </Button>
  <Button
    onClick={[Function]}
    style="clear"
  >
    Irreversibly shut down fund
  </Button>
</div>
`;
